R/read_csv.R

```diff
@@ -212,58 +244,94 @@ read_sample_csv <- function(output_files) {
       }
       id <- csv_file_info$id
     }
-    # read sampling data
-    draws <- utils::read.csv(output_file, header = TRUE, comment.char = "#")
+    repaired_model_params <- repair_variable_names(sampling_info$model_params)
+    repaired_sampler_diagnostics <- repair_variable_names(sampling_info$sampler_diagnostics)
+    if (!is.null(parameters)) {
+      for (p in parameters) {
+        if (!(p %in% repaired_model_params)) {
+          stop("The specified parameter ", p, " was not found in the sampling output!")
+        }
+      }
```

Comment:
I think it would be better to check these all at once instead of in a loop, so that if the user has multiple parameters that are not found they don't have to keep trying and finding out another one is wrong. 

What do you think? Something along these lines could work:

```r
not_found <- parameters[!parameters %in% repaired_model_params]
if (length(not_found)) {
  stop("Can't find parameter(s) ", paste(not_found, collapse = ", "), " in the sampling output!")
}
```

Reply:
Fixed.